.github/workflows/publish-package.yml

```
@@ -29,11 +29,41 @@ permissions:
   contents: read
 
 jobs:
-  build:
+  bump:
+    runs-on: ubuntu-latest
+    outputs: 
+      packages: ${{ steps.changesets.outputs.publishedPackages }}
 
+    steps:
+    - uses: pnpm/action-setup@v4
+    - name: Use Node.js 20
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20
+        cache: "pnpm"
+
+    - name: Install pnpm dependencies
+      run: pnpm install
+
+    - name: Create Release Pull Request
+      id: changesets
+      uses: changesets/action@v1
+      with:
+        version: pnpm ci:version
+        publish: pnpm ci:publish
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  
+  build:
+    needs:
+    - bump
+    strategy:
+      matrix:
+        package: ${{ fromJson(needs.bump.outputs.packages) }}
```

not sure if this will work, the reference calls for JSON to be capitalized
```suggestion
        package: ${{ fromJSON(needs.bump.outputs.packages) }}
```

---

https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson

---

no it will, this is the canonical name for it actually, github documentation just doesn't use it